java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/utils/HazelcastUtils.java

```diff
@@ -53,6 +60,36 @@ public static boolean isOldestMember(HazelcastInstance hazelcastInstance) {
         return memberIterator.hasNext() && memberIterator.next().equals(hazelcastInstance.getLocalEndpoint());
     }
 
+    /**
+     * Handler to set the member configuration with specific properties which
+     * are bound to an agent by internal IP.
+     * <p>
+     *     Sets CP priority for member based on the agent's private address
+     *     if provided.
+     * </p>
+     * @param properties
+     * @param config
+     */
+    public static void handlePerAgentConfig(Map<String, String> properties, Config config) {
+        String cpPriorities = properties.get("cp_priority");
+        if (cpPriorities == null) {
+            return;
+        }
+
+        String agentPrivateAddress = properties.get("PRIVATE_ADDRESS");
+        JSONArray cpPriorityArray = new JSONArray(cpPriorities);
+
+        cpPriorityArray.forEach(item -> {
+            JSONObject jsonObject = (JSONObject) item;
+            String address = jsonObject.getString("address");
+            int priority = jsonObject.getInt("priority");
+            if (address.equals(agentPrivateAddress)) {
+                LOGGER.info("Setting CP member priority to " + priority + " for agent " + agentPrivateAddress);
+                config.getCPSubsystemConfig().setCPMemberCount(priority);
```

Comment:
`setCPMemberPriority`

Reply:
Lol I don't know how this happened -- I'm gonna blame copilot though

Reply:
Thanks for spotting